### PR TITLE
SqliteSnapshotStore is not getting table name from config

### PR DESCRIPTION
**Pull request by maratoss**
Table name was hardcoded. Seems that is breaking changes :(

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Should close #4080

**Comment by Aaronontheweb** [comment]
Tests look good - will review soon

File: src/contrib/persistence/Akka.Persistence.Sqlite/Snapshot/ConflictNameResolver.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /// </summary>
+    internal class ConflictNameResolver
+    {
+        /// <summary>
+        /// Tries to resolve the conflict of snapshot table names
+        /// </summary>
+        /// <param name="connection">sqlite connection</param>
+        /// <param name="configuration">sqlite query configuration</param>
+        /// <returns>Returns true if has the conflict</returns>
+        public async Task<bool> Resolve(SqliteConnection connection, SqliteQueryConfiguration configuration)
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Sqlite/Snapshot/ConflictNameResolver.cs]: ... Why is this class necessary? I don't understand the need for the rest of the changes in this PR. I thought it was just a HOCON fallback?

Comment (maratoss) [reply src/contrib/persistence/Akka.Persistence.Sqlite/Snapshot/ConflictNameResolver.cs]: Did not get the point with HOCON fallback :)
The idea was to do not get db corrupted for people who upgrade akka library.
This fix gets akka still use old hardcoded table 'snapshot' despite of 'table-name' setting value.

Comment (Aaronontheweb) [reply src/contrib/persistence/Akka.Persistence.Sqlite/Snapshot/ConflictNameResolver.cs]: I'll take another look at it.

**Pull request closed without merging**